### PR TITLE
fix(kai-kli): make register process async

### PR DESCRIPTION
**Pull request by JYisus**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [ ] I have created tests for my code changes, and the tests are passing.
- [ ] I have executed the pre-commit hooks locally.
- [ ] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by alexrodfe — APPROVED** [review]

File: cmd/processregistry/process_registry.go [hunk 1]
```diff
@@ -19,6 +19,9 @@ func NewProcessRegistryCmd(logger logging.Interface) *cobra.Command {
 		`),
 	}
 
+	cmd.PersistentFlags().StringP(_serverFlag, "s", "", "KAI server to use")
+	_ = cmd.MarkPersistentFlagRequired(_serverFlag)
```
Comment (dfinteligenz) [inline cmd/processregistry/process_registry.go:23]: The server is not always required, if there is no server flag defined, the default server should be used.

File: internal/render/renderer.go [hunk 2]
```diff
@@ -163,14 +163,14 @@ func (r *CliRenderer) RenderConfiguration(scope string, config map[string]string
...
-func (r *CliRenderer) RenderRegisteredProcesses(registeredProcesses []processregistry.RegisteredProcess) {
+func (r *CliRenderer) RenderRegisteredProcesses(registeredProcesses []*processregistry.RegisteredProcess) {
 	if len(registeredProcesses) < 1 {
 		r.logger.Info("No processes found.")
 		return
 	}
 
 	r.tableWriter.SetHeader([]string{
-		"Process Name", "Process Version", "Process type", "Image", "Upload Date", "Owner",
+		"Name", "Version", "Type", "Image", "Upload Date", "Owner", "Status",
```
Comment (dfinteligenz) [inline internal/render/renderer.go:173]: I'd move the STATUS column just before the VERSION.

```suggestion
		"Name", "Version", "Status", "Type", "Image", "Upload Date", "Owner",
```

**Review by dfinteligenz — DISMISSED** [review]

**Review by alexrodfe — DISMISSED** [review]
Alright :+1:

File: cmd/processregistry/process_registry.go (hunk 1 above)
Comment (dfinteligenz) [reply cmd/processregistry/process_registry.go]: @JYisus check this comment please, everything else seems good to me!

**Review by dfinteligenz — APPROVED** [review]

**Pull request merged**